openapi_tester/loaders.py

```diff
@@ -128,34 +130,54 @@ def resolve_path(self, endpoint_path: str) -> tuple:
         parsed_path = url_object.path if url_object.path.endswith("/") else url_object.path + "/"
         if not parsed_path.startswith("/"):
             parsed_path = "/" + parsed_path
+        for key, value in self.field_key_map.items():
+            if value != "pk" and key in parsed_path:
+                parsed_path = parsed_path.replace(f"{{{key}}}", value)
         for path in [parsed_path, parsed_path[:-1]]:
             try:
                 resolved_route = resolve(path)
+            except Resolver404:
+                continue
+            else:
+                if "{pk}" in path and api_settings.SCHEMA_COERCE_PATH_PK:
+                    path, resolved_route = self.handle_pk_parameter(
+                        resolved_route=resolved_route, path=path, method=method
+                    )
                 for key, value in resolved_route.kwargs.items():
                     # Replacing kwarg values back into the string seems to be the simplest way of bypassing complex
                     # regex handling. However, its important not to freely use the .replace() function, as a
                     # {value} of `1` would also cause the `1` in api/v1/ to be replaced
                     var_index = path.rfind(str(value))
                     path = path[:var_index] + f"{{{key}}}" + path[var_index + len(str(value)) :]
                 return path, resolved_route
-            except Resolver404:
-                continue
         message = f"Could not resolve path `{endpoint_path}`."
-        closest_matches = "".join(
-            f"\n- {i}" for i in difflib.get_close_matches(endpoint_path, self.get_endpoint_paths())
-        )
+        closest_matches = "".join(f"\n- {i}" for i in difflib.get_close_matches(endpoint_path, self.endpoints))
         if closest_matches:
             message += f"\n\nDid you mean one of these?{closest_matches}"
         raise ValueError(message)
 
+    @staticmethod
+    def handle_pk_parameter(resolved_route: ResolverMatch, path: str, method: str) -> Tuple[str, ResolverMatch]:
+        split_view_path = resolved_route.view_name.split(".")
+        view_name = split_view_path.pop()
+        imported_module = import_module(".".join(split_view_path))
+        view = getattr(imported_module, view_name)
+        coerced_path = BaseSchemaGenerator().coerce_path(path=path, method=method, view=view)
```

Comment:
```suggestion
        model = getattr(resolved_route.func.view_class, 'model', None)
        field_name = get_pk_name(model) if model else 'id'
```
Drawing on the `coerce_path` method logic I think we might be able to shorten it to ~this. Unless there are cases where the `view_name` is included in a resolved parth, but `view_class` isn't; but that seems unlikely